### PR TITLE
Implement Ppro unified feedback

### PR DESCRIPTION
**Pull request by karlenDimla**
Task/Issue URL: https://app.asana.com/0/0/1207815400172108/f

### Description
See attached task description

### Steps to test this PR
https://app.asana.com/0/1201462763415876/1207962930855840/f

### CODE REVIEW

**Comment by karlenDimla** [comment]
* **#4840** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4840?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karlenDimla and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/FeedbackSubCategoryProvider.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface FeedbackSubCategoryProvider {
+    fun getSubCategories(category: SubscriptionFeedbackCategory): Map<Int, SubscriptionFeedbackSubCategory>
+}
+
+@ContributesBinding(FragmentScope::class)
+class RealFeedbackSubCategoryProvider @Inject constructor() : FeedbackSubCategoryProvider {
+    override fun getSubCategories(category: SubscriptionFeedbackCategory): Map<Int, SubscriptionFeedbackSubCategory> {
```
Comment (marcosholgado) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/FeedbackSubCategoryProvider.kt:34]: Tests?

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import logcat.logcat
+
+@ContributesViewModel(ActivityScope::class)
+class SubscriptionFeedbackViewModel @Inject constructor() : ViewModel() {
```
Comment (marcosholgado) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt:57]: Tests?

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+                REPORT_PROBLEM -> sendReportIssuePixel(metadata)
+                null -> {} // Do nothing
+            }
+
+            command.send(FeedbackCompleted)
+        }
+    }
+
+    private fun sendReportIssuePixel(metadata: FeedbackMetadata) {
+        logcat { "KLDIMSUM: sendReportIssuePixel for $metadata" }
```
Comment (marcosholgado) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt:140]: nit: change to a todo

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+    }
+
+    private fun sendFeatureRequestPixel(metadata: FeedbackMetadata) {
+        logcat { "KLDIMSUM: sendFeatureRequestPixel for $metadata" }
```
Comment (marcosholgado) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt:144]: nit: change to a todo

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+    }
+
+    private fun sendGeneralFeedbackPixel(metadata: FeedbackMetadata) {
+        logcat { "KLDIMSUM: sendGeneralFeedbackPixel for $metadata" }
```
Comment (marcosholgado) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/feedback/SubscriptionFeedbackViewModel.kt:148]: nit: change to a todo

**Review by marcosholgado — APPROVED** [review]
LGTM

**Pull request merged**